per/task.py

```diff
@@ -0,0 +1,31 @@
+from celery import shared_task
+
+from api.logger import logger
+from per.models import OpsLearningCacheResponse
+from per.ops_learning_summary import OpsLearningSummaryTask
+
+
+@shared_task
+def generate_summary(ops_learning_summary_id: int, filter_data: dict):
+    try:
+        ops_learning_summary_instance = OpsLearningCacheResponse.objects.filter(id=ops_learning_summary_id).first()
```

Comment:
move this outside try
```suggestion
        ops_learning_summary_instance = OpsLearningCacheResponse.objects.filter(id=ops_learning_summary_id).first()
        if ops_learning_summary_instance is None:
            # TODO: Add extra context for which ops learning cache response failed
            logger.error("Missing Ops learning cache response triggered")
            return False
```